### DISCUSSION TITLE
`IResult` vs `PResult`

### DISCUSSION THREAD

**Discussion by 9999years** [winnow-rs/winnow#327] [Q&A] [answered]
The documentation for `IResult` and `PResult` is very similar and doesn't explain how they're different. Practically I can see that `IResult` contains the remaining input in its `Ok` branch, but I'm not sure when this is used or if I should return it from my parsers (I'm assuming no because the tutorials use `PResult`).

Also, I'm not sure what the mnemonic for `IResult` is. I can tell that `PResult` is a parser result (the docs could spell this out, too), but is `IResult` an intermediate result? Input result, for its `InputError`?

Finally, the docs for `PResult` say that "By default, the error type (`E`) is `ErrorKind`", but the actual error type is `ContextError`. (The tutorials don't reflect this either.)

**Comment by epage** [comment, marked as answer]
I've tried to cover the answer for this in #328

**Reply by 9999years** [reply]
Thanks so much, that's super helpful!
